server/src/main/java/org/elasticsearch/common/blobstore/fs/FsBlobContainer.java

```diff
@@ -202,7 +203,7 @@ public void writeBlob(String blobName, InputStream inputStream, long blobSize, b
             if (failIfAlreadyExists) {
                 throw faee;
             }
-            deleteBlobsIgnoringIfNotExists(Collections.singletonList(blobName));
+            deleteBlobsIgnoringIfNotExists(Collections.singletonList(blobName).iterator());
```

Comment:
nit: Maybe we can extract this single item iterator pattern to `org.elasticsearch.common.collect.Iterators` we use it in a few places.

Reply:
++ makes sense